splinepy/helpme/create.py

```diff
@@ -277,6 +284,131 @@ def from_bounds(parametric_bounds, physical_bounds):
+            f"Dimension of parameter space and physical space ({self.whatami})"
+            "are not the same!"
+        )
+
+    # checks if spline is rational and if weights differ
+    # otherwise NURBS can be treated like regular BSpline
+
+    if self.is_rational and np.allclose(self.weights, self.weights.ravel()[0]):
+        rational_spline = True
+        log.debug(
+            f"Spline is rational ({self.whatami}) with unequal weights!"
+            "Determinant Spline will be calculated with all weights set to 1."
+        )
+        # save old weights
+        old_weights = self.weights
+        # overwrite weights of spline
+        self.weights = np.repeat([1], len(old_weights))
+    else:
+        rational_spline = False
+
+    # calculate degrees of det spline
+    deg_det = self.degrees * self.dim - 1
+    d = self.para_dim
+    # initialize lists for knot vectors
+    kvs_det = [None] * d
+    # calculate necessary additional knot multiplicity due to degree elevation
+    mult_inc = self.degrees * (d - 1) - 1
+
+    # knot_vectors
+    if self.has_knot_vectors:
+        for i in range(d):
+            # Add knots ([inner knots] * (mult_inc + 1))
+            # since continuity at inner knots further decreases by 1!
+            kvs_temp = np.array(self.knot_vectors[i]).tolist()
+            unique_knots = np.unique(kvs_temp).tolist()
+            kvs_temp += np.repeat(unique_knots[1:-1], mult_inc[i] + 1).tolist()
+            kvs_temp += np.repeat(
+                [unique_knots[0], unique_knots[-1]], mult_inc[i]
+            ).tolist()
+            kvs_temp.sort()
+            kvs_det[i] = kvs_temp
+
+        # number of cpts
+        n_cpts = np.prod([len(kvs_det[i]) - deg_det[i] - 1 for i in range(d)])
+
+        # create spline with empty cpts for calculation
+        determinant_projection = BSpline(
+            degrees=deg_det,
+            control_points=np.empty((n_cpts, 1)),
+            knot_vectors=kvs_det,
+        )
+
+    else:
+        # if Bezier Spline
+        n_cpts = np.prod([deg_det[i] + 1 for i in range(d)])
+
+        determinant_projection = Bezier(
+            degrees=deg_det, control_points=np.empty((n_cpts, 1))
+        )
+
+    # get det(J) of spline at greville pts to calculate cpts of det Spline
+    sample_points = determinant_projection.greville_abscissae(
+        duplicate_tolerance=1e-5
+    )
+    jac_dets = np.linalg.det(self.jacobian(sample_points))
+    basisfct_eval, supports = determinant_projection.basis_and_support(
+        sample_points
+    )
+
+    # build and solve system with scipy.sparse if available
+    if has_scipy:
+        n_row, n_col = supports.shape
+        row_ids = np.arange(n_row).repeat(n_col)
+        col_ids = supports.ravel()
+        data = basisfct_eval.ravel()
+        basis_functions = csr_array(
+            (data, (row_ids, col_ids)), shape=(n_row, n_row)
+        )
+        determinant_projection.control_points[:, 0] = linalg.spsolve(
```

Comment:
```suggestion
        determinant_projection.control_points[:] = linalg.spsolve(
```

Reply:
this unfortunately doesn't work:
```
ValueError: could not broadcast input array from shape (27,) into shape (27,1)
```
The issue is the same even when I create the control points with 
```
np.empty([n_cpts, 1])
```
so the issue is in the return of the `spline.control_points` function i guess?